system/jlib/jexcept.cpp

```diff
@@ -1354,7 +1360,10 @@ NO_SANITIZE("alignment") void excsighandler(int signum, siginfo_t *info, void *e
         if ( SEHHandler->fireException(new CSEHException(signum,s.str())) )
             return;
     }
-    raise(signum);
+    if ( (SIGABRT == signum) && (processAborted) )
+        _exit(255);
```

Comment:
perhaps should be setProcessAbortedCode , and return that code here instead of 255 ?

Reply:
We have to exit at this point and the init wrapper will not restart Thor for some exit values.

Reply:
sure but this in jlib, not Thor specific.
Thor's use can still use a code appropriate to it, e.g. 1

Reply:
Default from before would be 128 + SIGABRT (6) so thats 134.
Are you suggesting:
```
_exit(processAborted);
```

Reply:
So before it was 134 and it caused the Thor init scripts recycled(?)
As it stands it's hard-coded as 255 , and presumably that works with the thor init scripts and it still recycles?
I haven't checked the init script, but if 255 is what seemed appropriate, then you can have the calls to setProcessAbortCode set it to 255, in Thor code,
and then yes, here in jlib do:
```
_exit(processAbortCode)
```

Reply:
Yes, 134 and 255 both would cause recycle.